src/radio_instance.c

```diff
@@ -121,22 +130,35 @@ typedef struct radio_interface_desc {
 
```

Comment:
Another suggestion by @mlehtima (hopefully the last one)
```c
#define RADIO_INTERFACE_INDEX(x) (RADIO_INTERFACE_COUNT - x - 1)
```
and then hardcoded indices can be replaces with macros like this:
```c
static const RadioInterfaceDesc radio_interfaces[] = {
    {
        RADIO_INTERFACE_1_4,
        RADIO_1_4,
        radio_indication_ifaces + RADIO_INTERFACE_INDEX(RADIO_INTERFACE_1_4),
        radio_response_ifaces + RADIO_INTERFACE_INDEX(RADIO_INTERFACE_1_4)
    },{
        RADIO_INTERFACE_1_3,
        RADIO_1_3,
        radio_indication_ifaces + RADIO_INTERFACE_INDEX(RADIO_INTERFACE_1_3),
        radio_response_ifaces + RADIO_INTERFACE_INDEX(RADIO_INTERFACE_1_3)
    },{
        RADIO_INTERFACE_1_2,
        RADIO_1_2,
        radio_indication_ifaces + RADIO_INTERFACE_INDEX(RADIO_INTERFACE_1_2),
        radio_response_ifaces + RADIO_INTERFACE_INDEX(RADIO_INTERFACE_1_2)
    },{
        RADIO_INTERFACE_1_1,
        RADIO_1_1,
        radio_indication_ifaces + RADIO_INTERFACE_INDEX(RADIO_INTERFACE_1_1),
        radio_response_ifaces + RADIO_INTERFACE_INDEX(RADIO_INTERFACE_1_1)
    },{
        RADIO_INTERFACE_1_0,
        RADIO_1_0,
        radio_indication_ifaces + RADIO_INTERFACE_INDEX(RADIO_INTERFACE_1_0),
        radio_response_ifaces + RADIO_INTERFACE_INDEX(RADIO_INTERFACE_1_0)
    }
};
```
I would go even further and reduce it to something like this:
```c
#define RADIO_INTERFACE_DESC(v) \
        RADIO_INTERFACE_##v, RADIO_##v, \
        radio_indication_ifaces + RADIO_INTERFACE_INDEX(RADIO_INTERFACE_##v), \
        radio_response_ifaces + RADIO_INTERFACE_INDEX(RADIO_INTERFACE_##v)

static const RadioInterfaceDesc radio_interfaces[] = {
   { RADIO_INTERFACE_DESC(1_4) },
   { RADIO_INTERFACE_DESC(1_3) },
   { RADIO_INTERFACE_DESC(1_2) },
   { RADIO_INTERFACE_DESC(1_1) },
   { RADIO_INTERFACE_DESC(1_0) }
};
```

Reply:
After a few layers of macros, the c code starts to look like magic :)
Applied your suggestion.